File: src/test/scala/datomicScala/client/api/async/AsyncTxReportTest.scala
package datomicScala.client.api.async


class AsyncTxReportTest extends AsyncSetupSpec {
  sequential

  "4 txReport ops" in new AsyncSetup {

    films(txReport.dbBefore) == Nil
    films(txReport.dbAfter) == threeFilms

    // Tx datom + 3 entities * 3 attributes transacted
    txReport.txData.count === 1 + 3 * 3

    if (isDevLocal) {
      // No temp ids created with dev-local setup
      txReport.tempIds.size === 0
    } else {
      txReport.tempIds.size === 3
    }
  }
}

File: src/test/java/datomicJava/client/api/async/AsyncTxReportTest.java
package datomicJava.client.api.async;

import datomicJava.SetupAsync;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.collection.IsEmptyCollection.empty;


public class AsyncTxReportTest extends SetupAsync {

    public AsyncTxReportTest(String name) {
        system = name;
    }

    @Test
    public void txReportOps() throws ExecutionException, InterruptedException {
        assertThat(films(txReport.dbBefore()), is(empty()));
        assertThat(films(txReport.dbAfter()), is(threeFilms));

        // Tx datom + 3 entities * 3 attributes transacted
        assertThat(txReport.txData().count(), is((long) 1 + 3 * 3));

        if (isDevLocal()) {
            // No temp ids created with dev-local setup
            assertThat(txReport.tempIds().size(), is(0));
        } else {
            assertThat(txReport.tempIds().size(), is(3));
        }
    }
}